diff '--color=auto' -ur a/src/r8126_ptp.c b/src/r8126_ptp.c
--- a/src/r8126_ptp.c	2025-01-15 15:48:21.000000000 +0100
+++ b/src/r8126_ptp.c	2025-05-29 13:58:52.652475100 +0200
@@ -357,7 +357,7 @@
 }
 
 int rtl8126_get_ts_info(struct net_device *netdev,
-                        struct ethtool_ts_info *info)
+                        struct rtl_ethtool_ts_info *info)
 {
         struct rtl8126_private *tp = netdev_priv(netdev);
 
@@ -741,8 +741,12 @@
 
         /* init a hrtimer for pps */
         tp->pps_enable = 0;
+        #if LINUX_VERSION_CODE < KERNEL_VERSION(6,15,0)
         hrtimer_init(&tp->pps_timer, CLOCK_MONOTONIC, HRTIMER_MODE_REL);
         tp->pps_timer.function = rtl8126_hrtimer_for_pps;
+        #else
+        hrtimer_setup(&tp->pps_timer, rtl8126_hrtimer_for_pps, CLOCK_MONOTONIC, HRTIMER_MODE_REL);
+        #endif
 
         /* reset the PTP related hardware bits */
         rtl8126_ptp_reset(tp);
diff '--color=auto' -ur a/src/r8126_ptp.h b/src/r8126_ptp.h
--- a/src/r8126_ptp.h	2025-01-15 15:48:21.000000000 +0100
+++ b/src/r8126_ptp.h	2025-05-29 13:49:47.423912592 +0200
@@ -35,12 +35,19 @@
 #ifndef _LINUX_R8126_PTP_H
 #define _LINUX_R8126_PTP_H
 
+#include <linux/version.h>
 #include <linux/ktime.h>
 #include <linux/timecounter.h>
 #include <linux/net_tstamp.h>
 #include <linux/ptp_clock_kernel.h>
 #include <linux/ptp_classify.h>
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6,11,0)
+#  define rtl_ethtool_ts_info   ethtool_ts_info
+#else
+#  define rtl_ethtool_ts_info   kernel_ethtool_ts_info
+#endif /* LINUX_VERSION_CODE < KERNEL_VERSION(6,9,0) */
+
 #ifndef PTP_CLASS_NONE
 #define PTP_CLASS_NONE  0x00
 #endif
@@ -186,7 +193,7 @@
 struct RxDescV3;
 
 int rtl8126_get_ts_info(struct net_device *netdev,
-                        struct ethtool_ts_info *info);
+                        struct rtl_ethtool_ts_info *info);
 
 void rtl8126_ptp_reset(struct rtl8126_private *tp);
 void rtl8126_ptp_init(struct rtl8126_private *tp);
