diff '--color=auto' -ur a/src/r8126_ptp.c b/src/r8126_ptp.c
--- a/src/r8126_ptp.c	2025-04-26 06:57:43.987721811 +0200
+++ b/src/r8126_ptp.c	2025-04-26 07:08:32.491634949 +0200
@@ -357,7 +357,7 @@
 }
 
 int rtl8126_get_ts_info(struct net_device *netdev,
-                        struct ethtool_ts_info *info)
+                        struct rtl_ethtool_ts_info *info)
 {
         struct rtl8126_private *tp = netdev_priv(netdev);
 
diff '--color=auto' -ur a/src/r8126_ptp.h b/src/r8126_ptp.h
--- a/src/r8126_ptp.h	2025-04-26 06:57:43.987721811 +0200
+++ b/src/r8126_ptp.h	2025-04-26 07:06:24.789941239 +0200
@@ -35,12 +35,19 @@
 #ifndef _LINUX_R8126_PTP_H
 #define _LINUX_R8126_PTP_H
 
+#include <linux/version.h>
 #include <linux/ktime.h>
 #include <linux/timecounter.h>
 #include <linux/net_tstamp.h>
 #include <linux/ptp_clock_kernel.h>
 #include <linux/ptp_classify.h>
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6,11,0)
+#  define rtl_ethtool_ts_info   ethtool_ts_info
+#else
+#  define rtl_ethtool_ts_info   kernel_ethtool_ts_info
+#endif /* LINUX_VERSION_CODE < KERNEL_VERSION(6,9,0) */
+
 #ifndef PTP_CLASS_NONE
 #define PTP_CLASS_NONE  0x00
 #endif
@@ -186,7 +193,7 @@
 struct RxDescV3;
 
 int rtl8126_get_ts_info(struct net_device *netdev,
-                        struct ethtool_ts_info *info);
+                        struct rtl_ethtool_ts_info *info);
 
 void rtl8126_ptp_reset(struct rtl8126_private *tp);
 void rtl8126_ptp_init(struct rtl8126_private *tp);
